### PR TITLE
Add GitHubActionsTestLogger

### PR DESCRIPTION
**Pull request by Tyrrrz**
This will result in AppVeyor's output being a bit more noisy however. I don't know if that's an issue. If it is, I could improve the package to check if it actually runs on GHA or not.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 $ErrorActionPreference = "Stop"
 
-dotnet test -c Release `
+dotnet test -c Release -l GitHubActions `
```
Comment (bruno-garcia) [inline build.ps1:3]: Would be nice as you suggested for the package to auto detect, since folks (I for one) use these before pushing to remote

**Review by bruno-garcia — APPROVED** [review]

**Comment by Tyrrrz** [comment]
This is how it looks:

image

**Comment by Tyrrrz** [comment]
Should be good now. The build scripts will only use the logger when running on GHA and fallback to default console one locally and on travic/appveyor.

I had to bump target framework in Sentry.Protocols from net45 to net451. Didn't want to remove it outright as it seems like a bigger change to me, but we can do that in the future.

**Comment by bruno-garcia** [comment]
image

File: src/Sentry.Protocol/BaseScope.cs [hunk 2]
```diff
@@ -141,7 +141,7 @@ public User User
         /// An arbitrary mapping of additional metadata to store with the event.
         /// </summary>
         public
-#if NET45
+#if NET451
```
Comment (bruno-garcia) [inline src/Sentry.Protocol/BaseScope.cs:144]: Next versio will be 3.0 so feel free to add breaking changes although we should keep track of them.

On a separate PR we can drop anything older than net461

**Review by bruno-garcia — APPROVED** [review]

File: src/Sentry.Protocol/BaseScope.cs (hunk 2 above)
Comment (Tyrrrz) [reply src/Sentry.Protocol/BaseScope.cs:144]: I'm stupid, we only need to change this in tests, not on the actual project. Will update.

**Pull request merged**